### PR TITLE
Fix bug when pass undefined to function isEmpty and isNotEmpty

### PR DESCRIPTION
**Pull request by abraaoduarte**

### CODE REVIEW

File: lib/asserter.js [hunk 1]
```diff
@@ -105,71 +105,76 @@ class Assertion extends TestResultReporter {
...
   includesExactly(...objects) {
     const resultIsSuccessful = this._haveElementsConsideredEqual(this._actual, objects);
     const failureMessage = `${this.translated('include_exactly')} ${Utils.prettyPrint(objects)}`;
     this._reportAssertionResult(resultIsSuccessful, failureMessage);
   }
-  
-  isEmpty() {
-    const resultIsSuccessful = Utils.numberOfElements(this._actual) === 0;
+
+  isEmpty(value) {
```
Comment (ngarbezza) [inline lib/asserter.js:151]: this implementation of `isEmpty` is not expected to receive a value. It's called from line 70 on this same file. So, in other words, `value` is the same as `this._actual`

File: lib/asserter.js [hunk 2]
```diff
@@ -105,71 +105,76 @@ class Assertion extends TestResultReporter {
...
+    const setValueWhenUndefined = value || this._actual
+    const isUndefined = typeof value === undefined || this._actual === undefined;
+
+    const resultIsSuccessful = Utils.numberOfElements(setValueWhenUndefined || {}) === 0  && !isUndefined;
```
Comment (ngarbezza) [inline lib/asserter.js:155]: I would not prefer to do the comparison against `undefined` from the scope of this function. I rather delegate it to `Utils` module,  that has a `notNullOrUndefined` method. This has the advantage that your fix will work for both undefined and null

File: lib/asserter.js [hunk 3]
```diff
@@ -105,71 +105,76 @@ class Assertion extends TestResultReporter {
...
     const failureMessage = this.translated('be_empty');
     this._reportAssertionResult(resultIsSuccessful, failureMessage);
   }
-  
-  isNotEmpty() {
-    const resultIsSuccessful = Utils.numberOfElements(this._actual) > 0;
+
+  isNotEmpty(value) {
+    const setValueWhenUndefined = value || this._actual || {}
```
Comment (ngarbezza) [inline lib/asserter.js:161]: same comment than line 151 about the extra parameter. The value to check against is always `this._actual`

**Review by ngarbezza — CHANGES_REQUESTED** [review]
Thank you @abraaoduarte for the contribution!

But I think we have to address the extra `value` parameter before merging!

File: lib/asserter.js (hunk 1 above)
Comment (abraaoduarte) [reply lib/asserter.js]: Thanks for your comments. I made some changes according to your comment

File: lib/asserter.js (hunk 2 above)
Comment (abraaoduarte) [reply lib/asserter.js]: I used your function to check this

File: lib/asserter.js (hunk 3 above)
Comment (abraaoduarte) [reply lib/asserter.js]: I changed according to your comments

**Comment by ngarbezza** [comment]
@all-contributors please add @abraaoduarte for code! thank you so much

**Pull request merged**